From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Otto=20Pfl=C3=BCger?= <otto.pflueger@abscue.de>
Date: Mon, 9 Dec 2024 07:10:26 +0100
Subject: [PATCH 043/225] power: supply: sc27xx: allow reading the 'calibrate'
 property
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Properties which exist but are not readable cause a lot of unnecessary
errors. Make the 'calibrate' property readable to avoid them.

The value of this property is not important, so just return the
previously written value.

Signed-off-by: Otto Pflüger <otto.pflueger@abscue.de>
---
 drivers/power/supply/sc27xx_fuel_gauge.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/power/supply/sc27xx_fuel_gauge.c b/drivers/power/supply/sc27xx_fuel_gauge.c
index 580bd2cd2..842faa1b0 100644
--- a/drivers/power/supply/sc27xx_fuel_gauge.c
+++ b/drivers/power/supply/sc27xx_fuel_gauge.c
@@ -693,6 +693,10 @@ static int sc27xx_fgu_get_property(struct power_supply *psy,
 		val->intval = data->boot_volt;
 		break;
 
+	case POWER_SUPPLY_PROP_CALIBRATE:
+		val->intval = data->init_cap;
+		break;
+
 	default:
 		ret = -EINVAL;
 		break;
